pliers/support/decorators.py

```diff
@@ -19,3 +20,15 @@ def decorated(*args, **kwargs):
             print(err)
             raise MissingCorpusError()
     return decorated
+
+
+def requires_optional_dependency(dependency):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
```

Comment:
One danger in wrapping the entire function call is that if it fails with an `AttributeError` that's *not* due to the missing dependency, we'll be giving the user the wrong error message. So I think we need to rework this somehow. A couple of other possibilities:

* We could get rid of the function-level decorator and instead verify that the optional dependency is available at the class level--i.e., we only check once, in `init`. The downside of this is that it will fail even if the user only intends to call functions that *don't* need the offending library. However, this strikes me as unlikely, since in most cases the only interaction with `Transformer`s is through `transform`, so if the object can't be initialized without the requirement, it probably can't be used at all.

* Do it at function level, but make the checks inside the function rather than using a decorator. E.g., something like:

```python

def _extract(self):
    verify_dependencies_exist([sr])
    ...
```

...where the current function call just fails with an Exception if any of the listed dependencies evaluate to `None`. If we want to produce more informative exceptions, we could maintain a global dictionary of optional imports, where the variable name is the key, and the original module name is the value. I actually kind of like this solution, because in principle it would make it quite easy to track cases where *multiple* requirements are missing (and generate an informative message).

Reply:
What about just moving the `attempt_to_import` into the function and raise the error in there? Forgot if there was a reason to have the import attempts at the top.

Reply:
Main drawback I can see to that it is that it makes it hard to see that the import is happening at a glance, since it's not in the file head. Also, it's slightly more cumbersome code-wise, since we then need to assign the result of the call explicitly, so that instead of having something like `verify_dependencies([lib1, lib2])`, we have the more verbose `lib1, lib2 = attempt_to_import([lib1, lib2])`--possibly multiple times in each module. (We could have the assignment done implicitly inside `attempt_to_import`, but I think that's bad practice, as the assignment is then completely invisible).

On balance I still have a very slight preference towards the pattern I suggested, but if you prefer to do it this way, that's fine with me.